homeassistant/components/guardian/__init__.py

```diff
@@ -164,25 +389,14 @@ async def _async_internal_added_to_hass(self):
         """
         raise NotImplementedError
 
-    @callback
-    def _async_update_from_latest_data(self):
-        """Update the entity.
-
-        This should be extended by Guardian platforms.
-        """
-        raise NotImplementedError
-
     @callback
     def async_add_coordinator_update_listener(self, api: str) -> None:
         """Add a listener to a DataUpdateCoordinator based on the API referenced."""
-
-        @callback
-        def async_update():
-            """Update the entity's state."""
-            self._async_update_from_latest_data()
-            self.async_write_ha_state()
-
-        self.async_on_remove(self._coordinators[api].async_add_listener(async_update))
+        self.async_on_remove(
```

Comment:
We have a CoordinatorEntity mixin nowadays that does this for you.

Reply:
a335660. This feels like a case where multiple inheritance causes more trouble than it's worth...